src/call_builder.js

```diff
@@ -27,9 +28,14 @@ export class CallBuilder {
   checkFilter() {
     if (this.filter.length >= 2) {
       throw new BadRequestError("Too many filters specified", this.filter);
-    } 
+    }
+
     if (this.filter.length === 1) {
-      this.url.segment(this.filter[0]);
+      //append filters to original segments
+      let newSegment = this.originalSegments.concat(this.filter[0]);
+      forEach(this.filter[0], filter => {
+        this.url.segment(newSegment);
+      });
```

Comment:
I don't think this loop is needed here. You have new segments in `newSegment` so calling `this.url.segment(newSegment);` once should be enough.

Reply:
Fixed.

Reply:
Fixed.